adapters/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/core/JmsEndpointGeneratorTest.java

```diff
@@ -547,6 +549,51 @@ public void shouldCallJmsProcessorWhenOnMessageIsInvoked() throws Exception {
         verify(dispatcher).dispatch(envelope);
     }
 
+    @Test
+    public void shouldCreateDurableTopicSubscriber() throws Exception {
+        generator.run(raml()
+                        .withBaseUri("message://event/listener/message/people")
+                        .with(resource()
+                                .withRelativeUri("/people.events")
+                                .with(action(POST, "application/vnd.context1.events.abc+json")))
+                        .build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder));
+
+        Class<?> clazz = compiler.compiledClassOf(BASE_PACKAGE);
+        ActivationConfigProperty[] activationConfig = clazz.getAnnotation(MessageDriven.class).activationConfig();
+        assertThat(activationConfig, hasItemInArray(
+                allOf(propertyName(equalTo("destinationType")), propertyValue(equalTo("javax.jms.Topic")))));
+        assertThat(activationConfig, hasItemInArray(
+                allOf(propertyName(equalTo("subscriptionDurability")), propertyValue(equalTo("Durable")))));
+        assertThat(activationConfig, hasItemInArray(
+                allOf(propertyName(equalTo("clientId")), propertyValue(equalTo("people.event.listener")))));
+        assertThat(activationConfig, hasItemInArray(
+                allOf(propertyName(equalTo("subscriptionName")), propertyValue(equalTo("people.event.listener.people.events")))));
+
+    }
+
+    @Test
+    public void shouldNotContainDurableSubscriberPropertiesIfItsNotTopic() throws Exception {
+        generator.run(raml()
+                        .withBaseUri("message://command/controller/message/people")
+                        .with(resource()
+                                .withRelativeUri("/people.controller.commands")
+                                .with(action(POST, "application/vnd.people.events.abc+json")))
+                        .build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder));
+
+        Class<?> clazz = compiler.compiledClassOf(BASE_PACKAGE);
+        ActivationConfigProperty[] activationConfig = clazz.getAnnotation(MessageDriven.class).activationConfig();
+        assertThat(activationConfig, hasItemInArray(allOf(
+                propertyName(equalTo("destinationType")),
```

Comment:
tests should be verbose and descriptive, a form of documentation, you rarely use constants for stuff that is relevant to the test case, anything else goes to builder's defaults